### DISCUSSION TITLE
Giving up on Gluestack

### DISCUSSION THREAD

**Discussion by ameesme** [gluestack/gluestack-ui#2412] [General]
I've been a long-term user of GeekyAnts-frameworks, having built countless of projects with NativeBase and later Gluestack V1. I want to voice some concerns I have about the future of Gluestack.

After the "downfall" of NativeBase, I was eager to find a replacement. After using Tamagui for a while, I decided to give Gluestack a shot. The developer experience with unfortunately hasn't been great. In hindsight, the first "stable" release looked very rushed, with quite some major API changes during its development. Now, with the release of Gluestack V2 -that once again has a different API- I can no longer use or recommend it.

Developing with Gluestack has just been too volatile and unpredictable, as you can never predict the next time you have to refactor your application's UI stack. I can't bet the integrity of my and my client's apps on it any longer.

**Comment by brayohmurithidev** [comment]
So would you advice me to go with Tamagui? I have already tried Gluestack and it's not even starting

**Comment by sanketsahu** [comment]
@ameesme Sorry about your experience! We completely realize our mistakes. Moving from NativeBase v3 to gluestack v1 and now v2 wasn't easy for us either. I understand what it takes to start a project and then later upgrade it with some different APIs.

We tried our best to provide the right codemods whenever possible. With v2, we are going to tag it as stable and also come up with an LTS support tag which is under discussion.

The parts of the gluestack v2 stack are pretty stable.

1. NativeWind is seeing a 90k weekly downloads
2. The unstyled component library is pretty stable too.
3. The styling goes in the project so it's very easy to customize and stay with that without any conflicting styling from the parent library.

Happy to help you further and provide all the support! Thanks for believing in us for the longest time. We hope it continues!
